src/node_lttng.cc

```diff
@@ -0,0 +1,264 @@
+      (_##conn)->Get(FIXED_ONE_BYTE_STRING(env->isolate(), "_handle")); \
+  if (_handle->IsObject()) { \
+    SLURP_INT(_handle.As<Object>(), fd, &conn.fd); \
+  } else { \
+    conn.fd = -1; \
+  } \
+  SLURP_STRING(_##conn, remoteAddress, &conn.remote); \
+  SLURP_INT(_##conn, remotePort, &conn.port); \
+  SLURP_INT(_##conn, bufferSize, &conn.buffered);
+
+#define SLURP_CONNECTION_HTTP_CLIENT(arg, conn) \
+  if (!(arg)->IsObject()) { \
+    return env->ThrowError( \
+      "expected argument " #arg " to be a connection object"); \
+  } \
+  node_lttng_connection_t conn; \
+  Local<Object> _##conn = Local<Object>::Cast(arg); \
+  SLURP_INT(_##conn, fd, &conn.fd); \
+  SLURP_STRING(_##conn, host, &conn.remote); \
+  SLURP_INT(_##conn, port, &conn.port); \
+  SLURP_INT(_##conn, bufferSize, &conn.buffered);
+
+#define SLURP_CONNECTION_HTTP_CLIENT_RESPONSE(arg0, arg1, conn) \
+  if (!(arg0)->IsObject()) { \
+    return env->ThrowError( \
+      "expected argument " #arg0 " to be a connection object"); \
+  } \
+  if (!(arg1)->IsObject()) { \
+    return env->ThrowError( \
+      "expected argument " #arg1 " to be a connection object"); \
+  } \
+  node_lttng_connection_t conn; \
+  Local<Object> _##conn = Local<Object>::Cast(arg0); \
+  SLURP_INT(_##conn, fd, &conn.fd); \
+  SLURP_INT(_##conn, bufferSize, &conn.buffered); \
+  _##conn = Local<Object>::Cast(arg1); \
+  SLURP_STRING(_##conn, host, &conn.remote); \
+  SLURP_INT(_##conn, port, &conn.port);
+
+
+void LTTNG_NET_SERVER_CONNECTION(const FunctionCallbackInfo<Value>& args) {
+  if (!NODE_NET_SERVER_CONNECTION_ENABLED())
+    return;
+  Environment* env = Environment::GetCurrent(args);
+  SLURP_CONNECTION(args[0], conn);
+  NODE_NET_SERVER_CONNECTION(&conn, conn.remote, conn.port, conn.fd);
+}
+
+
+void LTTNG_NET_STREAM_END(const FunctionCallbackInfo<Value>& args) {
+  if (!NODE_NET_STREAM_END_ENABLED())
+    return;
+  Environment* env = Environment::GetCurrent(args);
+  SLURP_CONNECTION(args[0], conn);
+  NODE_NET_STREAM_END(&conn, conn.remote, conn.port, conn.fd);
+}
+
+
+void LTTNG_HTTP_SERVER_REQUEST(const FunctionCallbackInfo<Value>& args) {
+  node_lttng_http_server_request_t req;
+
+  if (!NODE_HTTP_SERVER_REQUEST_ENABLED())
+    return;
+
+  Environment* env = Environment::GetCurrent(args);
+  HandleScope scope(env->isolate());
+  Local<Object> arg0 = Local<Object>::Cast(args[0]);
+  Local<Object> headers;
+
+  memset(&req, 0, sizeof(req));
+  req._un.version = 1;
+  SLURP_STRING(arg0, url, &req.url);
+  SLURP_STRING(arg0, method, &req.method);
+  SLURP_OBJECT(arg0, headers, &headers);
+
+  if (!(headers)->IsObject()) {
+    return env->ThrowError(
+      "expected object for request to contain string member headers");
+  }
+
+  Local<Value> strfwdfor = headers->Get(env->x_forwarded_string());
+  node::Utf8Value fwdfor(env->isolate(), strfwdfor);
+
+  if (!strfwdfor->IsString() || (req.forwardedFor = *fwdfor) == nullptr)
+    req.forwardedFor = const_cast<char*>("");
```

Comment:
The struct member should (and can) be `const char*` because you're not mutating it after the fact.

Rule of thumb: if something can be const, it probably should.